tools/mot_visualization.py

```diff
@@ -0,0 +1,227 @@
+import argparse
+import os
+import os.path as osp
+import tempfile
+
+import cv2
+import mmcv
+import motmetrics as mm
+import numpy as np
+from mmcv import Config
+
+from mmtrack.datasets import build_dataset
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='visualize the situation of false positive, '
+        'false negative and ID switch for multiple object tracking')
+    parser.add_argument('config', help='test config file path')
+    parser.add_argument('--result-file', help='path of inference result')
+    parser.add_argument(
+        '--output', help='directory where painted images will be saved')
+    parser.add_argument(
+        '--out-video', action='store_true', help='whether output video')
+    parser.add_argument(
+        '--show',
+        action='store_true',
+        help='whether show the results on the fly')
+    parser.add_argument(
+        '--fps', type=int, default=30, help='FPS of the output video')
+    args = parser.parse_args()
+    return args
+
+
+def show_wrong_tracks(img,
+                      bboxes,
+                      ids,
+                      wrong_types,
+                      bbox_colors=[(0, 0, 255), (0, 255, 255), (255, 0, 0)],
+                      thickness=2,
+                      font_scale=0.4,
+                      show=False,
+                      wait_time=0,
+                      out_file=None):
+    """Show the wrong tracks with opencv.
+
+    red bbox means FP, yellow bbox means FN and blue bbox means IDSW
+    """
+    assert bboxes.ndim == 2
+    assert ids.ndim == 1
+    assert wrong_types.ndim == 1
+    assert bboxes.shape[1] == 5
+    assert len(bbox_colors) == 3
+    if isinstance(img, str):
+        img = mmcv.imread(img)
+
+    img_shape = img.shape
+    bboxes[:, 0::2] = np.clip(bboxes[:, 0::2], 0, img_shape[1])
+    bboxes[:, 1::2] = np.clip(bboxes[:, 1::2], 0, img_shape[0])
+
+    text_width, text_height = 10, 15
+    for bbox, wrong_type, id in zip(bboxes, wrong_types, ids):
+        x1, y1, x2, y2 = bbox[:4].astype(np.int32)
+        score = float(bbox[-1])
+
+        # bbox
+        bbox_color = bbox_colors[wrong_type]
+        cv2.rectangle(img, (x1, y1), (x2, y2), bbox_color, thickness=thickness)
+
+        # FN and IDSW do not need id and score
+        if wrong_type == 1 or wrong_type == 2:
+            continue
```

Comment:
IDSW needs to plot id and score. We need to know the id switch from which target.